### PR TITLE
Update data missing alert to use correct name

### PR DESCRIPTION
**Pull request by SeaJerry**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Updated the `NeedsMissingInfoAlert `component to display dynamic form names instead of always showing "Supplemental Claim request." The component now accepts a `formId` prop and uses a mapping function to show the correct form name: "Board Appeal request" for form 10182, "Supplemental Claim request" for 20-0995, and "Higher-Level Review request" for 20-0996.

Modified `ShowAlertOrSip` to pass the `formId` from `sipOptions` to the alert component, and updated the test spec to verify the new dynamic behavior works correctly for all form types while maintaining backwards compatibility.

## Related issue(s)

[Issue Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/112361)

## Testing done

Added tests to verify the dynamic form name functionality works correctly for each form ID: one test each for Board Appeal (10182), Supplemental Claim (20-0995), and Higher-Level Review (20-0996) to ensure they display their respective form names. We also added tests for edge cases including unknown form IDs and missing form IDs to verify the fallback behavior shows "appeal request."

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Mobile | Desktop |
| ------- | ------ | ----- |
| 10182  | Screenshot 2025-08-07 at 10 41 40 AM |  Screenshot 2025-08-07 at 8 10 36 AM |
| 20995 | Screenshot 2025-08-07 at 10 41 19 AM |  Screenshot 2025-08-07 at 8 09 41 AM |
| 20996 | Screenshot 2025-08-07 at 10 41 30 AM | Screenshot 2025-08-07 at 8 10 15 AM |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/components/NeedsMissingInfoAlert.jsx [hunk 1]
```diff
@@ -6,8 +6,20 @@ import recordEvent from 'platform/monitoring/record-event';
...
-const NeedsMissingInfoAlert = ({ missing }) => {
+// Helper function to get form name based on formId
+const getFormName = formId => {
+  const formNames = {
+    '10182': 'Board Appeal',
+    '20-0995': 'Supplemental Claim',
+    '20-0996': 'Higher-Level Review',
+  };
+
+  return formNames[formId] || 'appeal';
```
Comment (SeaJerry) [inline src/applications/appeals/shared/components/NeedsMissingInfoAlert.jsx:17]: I added this, but please let me know if I should remove it in case it doesn't align with content. Ideally, there should always be a form_id

**Review by gracexu94 — DISMISSED** [review]
LGTM

**Review by randimays — COMMENTED** [review]
One nitpick comment. Thanks Jerry!

**Review by randimays — APPROVED** [review]
